crypto/init.c

```diff
@@ -670,7 +670,8 @@ int OPENSSL_init_crypto(uint64_t opts, const OPENSSL_INIT_SETTINGS *settings)
     if ((opts & OPENSSL_INIT_ENGINE_OPENSSL)
             && !RUN_ONCE(&engine_openssl, ossl_init_engine_openssl))
         return 0;
-# if !defined(OPENSSL_NO_HW) && !defined(OPENSSL_NO_DEVCRYPTOENG)
+# if !defined(OPENSSL_NO_HW) && !defined(OPENSSL_NO_DEVCRYPTOENG) \
+     && !defined(OPENSSL_DEVCRYPTO_DYNAMIC)
```

Comment:
Same thing here...  (btw, I wonder why `OPENSSL_NO_HW` isn't checked up at line 310)

Reply:
Actually, the `no-hw` options have fallen out of use...  They were designed to indicate that the hardware isn't available and are supposed to work like an engine disabler, so one can regard `no-hw-{whatever}` as a synonym for `disable-{whatever}eng`, and `no-hw` as a synonym for `disable-engine`.  We should probably simply make it so instead of ignoring them for yet a bunch of years.  That's a matter of another PR, though...